GitCommands/Git/GitModule.cs

```diff
@@ -2143,42 +2143,47 @@ public IReadOnlyList<PatchFile> GetInteractiveRebasePatchFiles()
-                var isCurrentFound = false;
-                foreach (string todoCommit in todoCommits)
-                {
-                    if (todoCommit.StartsWith(commentChar))
-                    {
-                        continue;
-                    }
+            string commentChar = EffectiveConfigFile.GetString("core.commentChar", "#");
 
-                    string[] parts = todoCommit.Split(Delimiters.Space);
+            string[][] todoCommitsInfos = todoCommits.Where(l => !l.StartsWith(commentChar))
+                .Select(l => l.Split(Delimiters.Space))
+                .Where(p => p.Length >= 3)
+                .ToArray();
 
-                    if (parts.Length < 3)
-                    {
-                        continue;
-                    }
+            string? currentCommitShortHash = File.Exists(CurrentFilePath) ? File.ReadAllText(CurrentFilePath).Trim() : null;
+            var isCurrentFound = false;
 
-                    string commitHash = parts[1];
-                    CommitData? data = _commitDataManager.GetCommitData(commitHash, out var error);
-                    var isApplying = currentCommitShortHash is not null && commitHash.StartsWith(currentCommitShortHash);
-                    isCurrentFound |= isApplying;
+            RevisionReader reader = new(this, hasReflogSelector: false);
+            CancellationTokenSource cts = new(2 * 60_000);
+            Dictionary<string, GitRevision> rebasedCommitsRevisions =
+                reader.GetRevisionsFromRange(todoCommitsInfos[0][1], todoCommitsInfos[^1][1], cts.Token)
+                .ToDictionary(r => r.Guid, r => r);
 
-                    patchFiles.Add(new PatchFile
-                    {
-                        Author = error ?? data?.Author,
-                        ObjectId = data?.ObjectId,
-                        Subject = error ?? data?.Body,
-                        Action = parts[0],
-                        Date = error ?? data?.CommitDate.LocalDateTime.ToString(),
-                        IsNext = isApplying,
-                        IsApplied = !isCurrentFound,
-                    });
-                }
+            foreach (string[] parts in todoCommitsInfos)
+            {
+                string commitHash = parts[1];
+                CommitData? data = rebasedCommitsRevisions.TryGetValue(commitHash, out GitRevision commitRevision)
+                    ? _commitDataManager.CreateFromRevision(commitRevision, null)
+                    : _commitDataManager.GetCommitData(commitHash, out var _);
+
+                var isApplying = currentCommitShortHash is not null && commitHash.StartsWith(currentCommitShortHash);
+                isCurrentFound |= isApplying;
+
+                patchFiles.Add(new PatchFile
+                {
+                    Author = data?.Author,
+                    ObjectId = data?.ObjectId,
+                    Subject = data?.Body,
```

Comment:
Is body or Subject needed?
RevisionReader will not save Body (just subject) if the commit is older than 6 months. 
(rename the _sixMonths variable to _oldestBody?)

```
        public RevisionReader(GitModule module, bool hasReflogSelector, bool allBodies = false)
            : this(module, hasReflogSelector, module.LogOutputEncoding, allBodies ? 0 : new DateTimeOffset(DateTime.Now.ToUniversalTime() - TimeSpan.FromDays(30 * 6)).ToUnixTimeSeconds())
        {
        }
```

Reply:
```suggestion
                    Subject = error ?? data?.Subject,
```